src/Main.elm

```diff
@@ -45,24 +100,68 @@ view : Model -> Html Msg
+        , div [ class "display" ] [ text model.display ]
         , div [ class "buttons" ]
             [ button [ class "operator" ] [ text "+" ]
             , button [ class "operator" ] [ text "-" ]
             , button [ class "operator" ] [ text "/" ]
             , button [ class "operator" ] [ text "*" ]
-            , button [ class "number" ] [ text "1" ]
-            , button [ class "number" ] [ text "2" ]
-            , button [ class "number" ] [ text "3" ]
-            , button [ class "number" ] [ text "4" ]
-            , button [ class "number" ] [ text "5" ]
-            , button [ class "number" ] [ text "6" ]
-            , button [ class "number" ] [ text "7" ]
-            , button [ class "number" ] [ text "8" ]
-            , button [ class "number" ] [ text "9" ]
-            , button [ class "number" ] [ text "0" ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumOne)
+                ]
+                [ text <| buttonValue CalNumOne ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumTwo)
+                ]
+                [ text <| buttonValue CalNumTwo ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumThree)
+                ]
+                [ text <| buttonValue CalNumThree ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumFour)
+                ]
+                [ text <| buttonValue CalNumFour ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumFive)
+                ]
+                [ text <| buttonValue CalNumFive ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumSix)
+                ]
+                [ text <| buttonValue CalNumSix ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumSeven)
+                ]
+                [ text <| buttonValue CalNumSeven ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumEight)
+                ]
+                [ text <| buttonValue CalNumEight ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumNine)
+                ]
+                [ text <| buttonValue CalNumNine ]
+            , button
+                [ class "number"
+                , onClick (UpdateDisplay CalNumZero)
+                ]
+                [ text <| buttonValue CalNumZero ]
             , button [ class "clearAll" ] [ text "AC" ]
-            , button [ class "decimal" ] [ text "." ]
+            , button
+                [ class "decimal"
+                , onClick <| UpdateDisplay CalDecimal
+                ]
+                [ text <| buttonValue CalDecimal ]
```

Comment:
You could extract a `buttonView` function here to reduce the repetition of this button code